media-source-respec.html

```diff
@@ -829,19 +829,17 @@
           <p>Follow these steps when <a def-id="duration"></a> needs to change to a <var>new duration</var>.</p>
           <ol>
             <li>If the current value of <a def-id="duration"></a> is equal to <var>new duration</var>, then return.</li>
-            <li>Set <var>old duration</var> to the current value of <a def-id="duration"></a>.
+			<li>If <var>new duration</var> is less than the highest start time reported by the <a def-id="buffered"></a> attribute across all <a>SourceBuffer</a> 
+			objects in <a def-id="sourceBuffers"></a>, then throw an <a def-id="invalid-state-error"></a> exception and abort these steps.</li>
+			  <p class="note">Duration reductions must first use <a def-id="remove"></a> to reduce the buffered range before updating <a def-id="duration"></a>.</p>
+            <li>Set <var>old duration</var> to the current value of <a def-id="duration"></a>.</li>
             <li>Update <a def-id="duration"></a> to <var>new duration</var>.</li>
-            <li>If the <var>new duration</var> is less than <var>old duration</var>, then run the
-              <a def-id="range-removal"></a> algorithm with <var>new duration</var> and
-              <var>old duration</var> as the start and end of the removal range.
-
-              <p class="note">This preserves audio frames and text cues that start before and end after the <a def-id="duration"></a>.</p></li>
-            <li>If a user agent is unable to partially render audio frames or text cues that start before and end after the <a def-id="duration"></a>, then run the following steps:
+            <li>If a user agent is unable to partially render audio frames or text cues that start before and end after the <a def-id="duration"></a>, then run the following steps:</li>
+			  <p class="note">This condition can occur because the <a def-id="coded-frame-removal-algorithm"></a> preserves audio frames and text cues that start before and end after the <a def-id="duration"></a>.</p>
```

Comment:
We've not changed the **Coded Frame Removal Algorithm** and the issue comes from it.

The Coded Frame Removal Algorithm takes starting and end removal range timestamps, and removes any data with a *starting timestamp* inside the removal range.

> Remove all media data, from this track buffer, that contain starting timestamps greater than or equal to start and less than the remove end timestamp.
 
Data that starts before the removal range and ends during it is not removed.  This means:

- Ending timestamps past duration are possible
- This affects all data types
- I'm not sure what the specific audio frame/text cue issue is, and decided to accommodate the mitigation for it.

I checked the start time in **Duration Change** on purpose because of this.  Then I left the audio/text cue special case to bump the duration if necessary.